clients/src/main/java/org/apache/kafka/clients/consumer/internals/PartitionAssignor.java

```diff
@@ -186,6 +182,10 @@ public ByteBuffer userData() {
             return userData;
         }
 
+        public void setGroupInstanceId(Optional<String> groupInstanceId) {
```

Comment:
Turning a final field into a mutable field seems like a step backwards... Can you explain why this is needed?